OpenShift startup fails (on first start)

```
$ sudo systemctl enable openshift
$ sudo systemctl start openshift
Job for openshift.service failed because the control process exited with error code. See "systemctl status openshift.service" and "journalctl -xe" for details.

$ sudo journalctl --unit=openshift
Feb 15 09:21:36 localhost.localdomain systemd[1]: openshift.service: main process     exited, code=exited, status=143/n/a
Feb 15 09:21:36 localhost.localdomain systemd[1]: Failed to start Docker     Application Container for OpenShift.
Feb 15 09:21:36 localhost.localdomain systemd[1]: Unit openshift.service entered     failed state.
Feb 15 09:21:36 localhost.localdomain systemd[1]: openshift.service failed.
Feb 15 09:22:06 localhost.localdomain systemd[1]: openshift.service holdoff time     over, scheduling restart.
Feb 15 09:22:06 localhost.localdomain systemd[1]: Starting Docker Application     Container for OpenShift...
Feb 15 09:22:06 localhost.localdomain docker[13560]: openshift
Feb 15 09:22:07 localhost.localdomain docker[13565]: openshift
Feb 15 09:22:07 localhost.localdomain docker[13571]: Error response from daemon:     Conflict: Tag latest is already set to image     62208f15133798af253a651386ae2753da9130fcc68f1863cbd735c547356173, if you want to     replace it, please use -f opt
Feb 15 09:22:08 localhost.localdomain sh[13667]: [INFO] Starting OpenShift server    
```

Actually in openshift_vagrant we did something like this:

```
########################################################################
wait_for_openshift() {
  # Give OpenShift time to start
  for i in {1..6}
  do
    curl -ksSf https://${1}:8443/api > /dev/null 2>&1
    if [ $? -ne 0 ]; then
      sleep 5
    else
      break
    fi
  done

  # Final check whether OpenShift is running
  curl -ksSf https://{$1}:8443/api > /dev/null 2>&1
  if [ $? -ne 0 ]; then
    >&2 echo "[ERROR] OpenShift failed to start:"
    docker logs ose
    exit 1
  fi
}
```

That said, one would assume that the check whether openshift is running is happening in _/opt/adb/openshift/openshift_. There should really be no need anymore for this check once _openshift_provision_ is called.

BTW, I am skeptical about the use of 127.0.0.1. The internal OpenShift IP is 10.0.2.15.

--- a/services/openshift/scripts/openshift
+++ b/services/openshift/scripts/openshift
@@ -61,22 +61,6 @@ start_openshift() {
         openshift start "${@:2}"
 }
 
-wait_for_openshift_api() {
-    # Wait for container to show up
-    ATTEMPT=0
-    until docker inspect openshift > /dev/null 2>&1 || [ $ATTEMPT -eq 10 ]; do
-      sleep 1
-      ((ATTEMPT++))
-    done
-
-    # Wait for API to be accessible
-    ATTEMPT=0
-    until curl -ksSf https://$ip:8443/api > /dev/null 2>&1 || [ $ATTEMPT -eq 30 ]; do
-      sleep 1
-      ((ATTEMPT++))
-    done
-}
-
 ########################################################################
 # Helper function to remove existing openshift container
 ########################################################################
@@ -155,4 +139,3 @@ fi
 # Now we start the server pointing to the prepared config files
 echo "[INFO] Starting OpenShift server"
 start_openshift '' --master-config="${master_config}" --node-config="${node_config}"
-wait_for_openshift_api

--- a/services/openshift/scripts/openshift_provision
+++ b/services/openshift/scripts/openshift_provision
@@ -9,6 +9,24 @@ export ORIGIN_DIR="/var/lib/origin"
 export OPENSHIFT_DIR=${ORIGIN_DIR}/openshift.local.config/master
 export KUBECONFIG=${OPENSHIFT_DIR}/admin.kubeconfig
 
+# Function to check openshift availability
+wait_for_openshift_api() {
+    # Wait for container to show up
+    ATTEMPT=0
+    until docker inspect openshift > /dev/null 2>&1 || [ $ATTEMPT -eq 10 ]; do
+      sleep 1
+      ((ATTEMPT++))
+    done
+
+    # Wait for API to be accessible
+    ATTEMPT=0
+    until curl -ksSf https://127.0.0.1:8443/api > /dev/null 2>&1 || [ $ATTEMPT -eq 30 ]; do
+      sleep 1
+      ((ATTEMPT++))
+    done
+}
+
+wait_for_openshift_api
 # Final check whether OpenShift is running
 curl -ksSf https://127.0.0.1:8443/api > /dev/null 2>&1
 if [ $? -ne 0 ]; then

